streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -111,6 +111,12 @@ public StateStoreSupplier build() {
 
                                     @Override
                                     public PersistentKeyValueFactory<K, V> windowed(final long windowSize, final long retentionPeriod, final int numSegments, final boolean retainDuplicates) {
+                                        if (numSegments < 1) {
```

Comment:
Yes, i think it is good practice to do it here as it is thrown as early as possible. We also need to have it `RocksDBWindowStoreSupplier` as there is nothing to stop users from just constructing that class.